Owl/app/src/main/java/com/example/owl/ui/OwlApp.kt

```diff
@@ -17,38 +17,42 @@
 package com.example.owl.ui
 
 import androidx.activity.OnBackPressedDispatcher
-import androidx.compose.animation.Crossfade
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.Providers
 import androidx.compose.runtime.remember
-import androidx.compose.runtime.savedinstancestate.rememberSavedInstanceState
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import com.example.owl.ui.course.CourseDetails
 import com.example.owl.ui.courses.Courses
 import com.example.owl.ui.onboarding.Onboarding
 import com.example.owl.ui.utils.AmbientBackDispatcher
-import com.example.owl.ui.utils.Navigator
 import com.example.owl.ui.utils.ProvideDisplayInsets
 import com.example.owl.ui.utils.ProvideImageLoader
 
 @Composable
 fun OwlApp(backDispatcher: OnBackPressedDispatcher) {
-    @Suppress("RemoveExplicitTypeArguments")
-    val navigator: Navigator<Destination> = rememberSavedInstanceState(
-        saver = Navigator.saver<Destination>(backDispatcher)
-    ) {
-        Navigator(Destination.Onboarding, backDispatcher)
-    }
-    val actions = remember(navigator) { Actions(navigator) }
 
     Providers(AmbientBackDispatcher provides backDispatcher) {
         ProvideDisplayInsets {
             ProvideImageLoader {
-                Crossfade(navigator.current) { destination ->
-                    when (destination) {
-                        Destination.Onboarding -> Onboarding(actions.onboardingComplete)
-                        Destination.Courses -> Courses(actions.selectCourse)
-                        is Destination.Course -> CourseDetails(
-                            destination.courseId,
+                val navController = rememberNavController()
+
+                val actions = remember(navController) { Actions(navController) }
+                NavHost(navController = navController, startDestination = Destinations.Onboarding) {
+                    composable(Destinations.Onboarding) {
+                        Onboarding(actions.onboardingComplete)
+                    }
+                    composable(Destinations.Courses) {
+                        Courses(actions.selectCourse)
+                    }
+                    composable(
+                        Destinations.Courses + "/{${Destinations.CoursesArgs.CourseId}}"
+                    ) { backStackEntry ->
+                        val courseId = requireNotNull(backStackEntry.arguments)
+                            .getLong(Destinations.CoursesArgs.CourseId)
```

Comment:
You have to explicitly mention that this argument is of type Long, by passing it to the `composable` function. Otherwise it will be added as a String argument by default.